tools/client-plugins/browser-scripts/frame-runner.ts

```diff
@@ -92,7 +92,13 @@ async function initTestFrame(e: InitTestFrameArg = { code: {} }) {
         // frameDocument ready:
         $(() => {
           try {
-            const test: unknown = eval(testString);
+            let test: unknown;
+            if (__file('script.js').indexOf(editableContents) > -1) {
+              const script = editableContents;
```

Comment:
This will likely cause quite a few bugs. If we're only evaluating the editableContents, then any time we have the camper writing a function that uses a variable outside of scope, the tests won't be able to pass.

Reply:
For challenges like [Step 82 of the Role Playing Game](http://freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-82), it appears we already have the global variables. Though any variables scoped specifically inside a function were near impossible to get. 

[The existing source file](https://github.com/freeCodeCamp/freeCodeCamp/blob/08b58f5417350b9473adfb1ad47c148ffe65ace7/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8c89e4272512d44fc1c66.md)

I am referring to challenges like [Step 47 of the Todo App](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-localstorage-by-building-a-todo-app/step-47). 

Depending on how well isolation goes, we'll be able to properly access both the global functions and the locally scoped variables inside of functions without any issues.